src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+from typing import Optional, Union
+import numpy as np
+import networkx as ntx
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import PathPatch
+from matplotlib.path import Path
+
+
+class SchedulingProblem:
+    def __init__(self,
+                 num_agents: int = 3,
+                 num_tasks: int = 3,
+                 sat_cutoff: Union[float, int] = 0.99,
+                 seed: int = 42):
+        """Schedule `num_tasks` tasks among `num_agents` agents such that
+        every agent performs exactly one task and every task gets assigned
+        exactly one agent.
```

Comment:
'to'

Thanks for providing a docstring with a great structure!
Could you maybe extend it slightly? Based on this description, it does not become clear why it is difficult to just schedule num_tasks among num_agents, since no constraints or similar is described.